src/dataflow-types/src/logging.rs

```diff
@@ -189,6 +197,10 @@ impl LogVariant {
                 vec![(0, 0), (1, 1)],
             )],
             LogVariant::Timely(TimelyLog::Parks) => vec![],
+            LogVariant::Timely(TimelyLog::Messages) => vec![(
+                LogVariant::Timely(TimelyLog::Channels),
```

Comment:
this foreign key constraint seems maybe incorrect? the way I'm reading the code this is saying "Each row in messages must have a col0 (channel id) equal to the channel id of {some, exactly one} (I forget how foreign keys work) row in channels"? and then it says that twice?

I think if this is trying to reference the primary key of channels then we want logically

```
(messages.channel, messages.source_worker) = (channels.id, channels.worker)
```
as the fk constraint? but again, not totally sure whether foreign keys mean "an element with this key exists in the other relation" or "exactly one element with this key exists in the other relation".

Reply:
I suspect that might be related to why https://github.com/MaterializeInc/materialize/issues/5985 happens